packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1385,6 +1385,18 @@ function applyVirtualResolutionMutations({
     throw new ReportError(MessageName.STACK_OVERFLOW_RESOLUTION, `Encountered a stack overflow when resolving peer dependencies; cf ${logFile}`);
   };
 
+  const getPackageFromDescriptor = (descriptor: Descriptor): Package => {
```

Comment:
This just moved the method up so it could be used to retrieve the peer dependency package